fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,382 @@
+"""
+Finds leaks between splits.
+"""
+
+from copy import copy
+
+from fiftyone import ViewField as F
+import fiftyone.core.brain as fob
+import fiftyone.brain.similarity as sim
+import fiftyone.core.utils as fou
+import fiftyone.core.validation as fov
+
+fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+
+_DEFAULT_MODEL = "clip-vit-base32-torch"
+_DEFAULT_BATCH_SIZE = None
+
+
+def compute_leaky_splits(
+    samples,
+    brain_key=None,
+    split_views=None,
+    split_field=None,
+    split_tags=None,
+    threshold=0.2,
+    similarity_brain_key=None,
+    embeddings_field=None,
+    model=None,
+    model_kwargs=None,
+    similarity_backend=None,
+    similarity_config_dict=None,
+    **kwargs,
+):
+    """See ``fiftyone/brain/__init__.py``."""
+
+    fov.validate_collection(samples)
+
+    config = LeakySplitsConfig(
+        split_views,
+        split_field,
+        split_tags,
+        similarity_brain_key,
+        embeddings_field,
+        model,
+        model_kwargs,
+        similarity_backend,
+        similarity_config_dict,
+        **kwargs,
+    )
+    brain_method = config.build()
+    brain_method.ensure_requirements()
+
+    if brain_key is not None:
+        # Don't allow overwriting an existing run with same key, since we
+        # need the existing run in order to perform workflows like
+        # automatically cleaning up the backend's index
+        brain_method.register_run(samples, brain_key, overwrite=False)
+
+    results = brain_method.initialize(samples, brain_key)
+
+    if results._save_similarity_index:
+        results._similarity_method.register_run(
+            samples,
+            results.config.similarity_brain_key,
+        )
+
+    results.set_threshold(threshold)
+    leaks = results.leaks
```

Comment:
Getting this error:

```
File ~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:856, in SimilarityIndex.get_model(self)
    [854](https://file+.vscode-resource.vscode-cdn.net/Users/jacobmarks/Desktop/~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:854) model = self.config.model
    [855](https://file+.vscode-resource.vscode-cdn.net/Users/jacobmarks/Desktop/~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:855) if model is None:
--> [856](https://file+.vscode-resource.vscode-cdn.net/Users/jacobmarks/Desktop/~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:856)     raise ValueError("These results don't have a stored model")
    [858](https://file+.vscode-resource.vscode-cdn.net/Users/jacobmarks/Desktop/~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:858) if etau.is_str(model):
    [859](https://file+.vscode-resource.vscode-cdn.net/Users/jacobmarks/Desktop/~/Desktop/work/brain/fiftyone-brain/fiftyone/brain/similarity.py:859)     model_kwargs = self.config.model_kwargs or {}

ValueError: These results don't have a stored model
```

When running this:

```py
dataset = foz.load_zoo_dataset("coco-2017", split="validation")
view1 = dataset.limit(1000)
view2 = dataset.skip(1000)

fob.compute_leaky_splits(dataset, split_views={"train":view1, "test":view2})
```

Reply:
Same thing happens when I add `similarity_brain_key="sim"`, and whether or not I include a `brain_key` arg

Reply:
I never considered the option of someone not passing a model at all. I'll add a default.

Reply:
Fixed how default values are handled, should work now. For what it's worth I highly recommend computing at least the embeddings ahead of time so it's batched and much much faster.